packages/fast-components-react-base/src/listbox/listbox.spec.tsx

```diff
@@ -0,0 +1,219 @@
+import * as React from "react";
+import * as Adapter from "enzyme-adapter-react-16";
+import { configure, mount, shallow } from "enzyme";
+import Listbox, { ListboxUnhandledProps } from "./listbox";
+import { KeyCodes } from "@microsoft/fast-web-utilities";
+
+/*
+ * Configure Enzyme
+ */
+configure({ adapter: new Adapter() });
+
+describe("context menu", (): void => {
+    test("should have a displayName that matches the component name", () => {
+        expect((Listbox as any).name).toBe(Listbox.displayName);
+    });
+
+    test("should not throw if managedClasses are not provided", () => {
+        expect(() => {
+            shallow(<Listbox />);
+        }).not.toThrow();
+    });
+
+    test("should implement unhandledProps", (): void => {
+        const unhandledProps: ListboxUnhandledProps = {
+            "aria-label": "label",
+        };
+
+        const rendered: any = shallow(<Listbox {...unhandledProps} />);
+
+        expect(rendered.first().prop("aria-label")).toEqual("label");
+    });
+
+    test("should not have a focusIndex if no children are focusable", (): void => {
+        const rendered: any = mount(
+            <Listbox>
+                <div>hello world</div>
+            </Listbox>
+        );
+
+        expect(rendered.state("focusIndex")).toBe(-1);
+    });
+
+    test("should set focusIndex to the first focusable element on mount", (): void => {
+        const rendered: any = mount(
+            <Listbox>
+                <div>not a focusable element</div>
+                <div role="option">focusable element</div>
+            </Listbox>
+        );
+
+        expect(rendered.state("focusIndex")).toBe(1);
+    });
+
+    test("should move focus down when the down arrow is pressed", (): void => {
+        const rendered: any = mount(
+            <Listbox>
+                <div role="option">one</div>
+                <div role="option">two</div>
+            </Listbox>
+        );
+
+        rendered.childAt(0).simulate("keydown", { keyCode: KeyCodes.arrowDown });
+
+        expect(rendered.state("focusIndex")).toBe(1);
```

Comment:
I would check before simulation that the focusIndex is 0

Reply:
yep, fix queued for next iteration